File: app/src/main/java/com/example/energo_monitoring/checks/data/db/OtherInfo.java
package com.example.energo_monitoring.checks.data.db;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity
public class OtherInfo {

    @PrimaryKey (autoGenerate = true)
    public int id;
    public int dataId;
    public int clientId;
    public int organizationId;
    public int projectId;
    public int userId;
    public int completedScreens;
    public int currentScreen;
    public int localVersion;
    public int cloudVersion;
    public boolean completed;

    public Boolean lightIsOk;
    public Boolean sanPinIsOk;
    public String generalInspectionComment;
    public String counterCharacteristicts;
    public String counterCharacteristictsComment;
    @Ignore
    public String finalPhotosGeneral;
    @Ignore
    public String finalPhotosDevices;
    @Ignore
    public String finalPhotosSeals;
    public String finalPhotosGeneralPath;
    public String finalPhotosDevicesPath;
    public String finalPhotosSealsPath;

    public OtherInfo() {
    }

    @Ignore
    public OtherInfo(int dataId) {
        this.dataId = dataId;
    }

    @Ignore
    public OtherInfo(int dataId, Boolean lightIsOk, Boolean sanPinIsOk, String generalInspectionComment) {
        this.dataId = dataId;
        this.lightIsOk = lightIsOk;
        this.sanPinIsOk = sanPinIsOk;
        this.generalInspectionComment = generalInspectionComment;
    }
}

File: app/src/main/java/com/example/energo_monitoring/checks/data/db/OtherInfo.kt
package com.example.energo_monitoring.checks.data.db

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.Ignore

@Entity
class OtherInfo {
    @JvmField
    @PrimaryKey(autoGenerate = true)
    var id = 0
    @JvmField
    var dataId = 0
    @JvmField
    var clientId = 0
    @JvmField
    var organizationId = 0
    @JvmField
    var projectId = 0
    @JvmField
    var userId = 0
    @JvmField
    var completedScreens = 0
    @JvmField
    var currentScreen = 0
    @JvmField
    var localVersion = 0
    @JvmField
    var cloudVersion = 0
    @JvmField
    var completed = false
    @JvmField
    var lightIsOk: Boolean? = null
    @JvmField
    var sanPinIsOk: Boolean? = null
    @JvmField
    var generalInspectionComment: String = ""
    @JvmField
    var counterCharacteristicts: String = ""
    @JvmField
    var counterCharacteristictsComment: String = ""

    @Ignore
    var finalPhotosGeneral: String = ""

    @Ignore
    var finalPhotosDevices: String = ""

    @Ignore
    var finalPhotosSeals: String = ""
    @JvmField
    var finalPhotosGeneralPath: String = ""
    @JvmField
    var finalPhotosDevicesPath: String = ""
    @JvmField
    var finalPhotosSealsPath: String = ""

    constructor() {}

    @Ignore
    constructor(dataId: Int) {
        this.dataId = dataId
    }

    @Ignore
    constructor(dataId: Int, lightIsOk: Boolean?, sanPinIsOk: Boolean?, generalInspectionComment: String) {
        this.dataId = dataId
        this.lightIsOk = lightIsOk
        this.sanPinIsOk = sanPinIsOk
        this.generalInspectionComment = generalInspectionComment
    }
}